src/qt/receivecoinsdialog.cpp

```diff
@@ -87,11 +87,18 @@ void ReceiveCoinsDialog::setModel(WalletModel *_model)
             &QItemSelectionModel::selectionChanged, this,
             &ReceiveCoinsDialog::recentRequestsView_selectionChanged);
 
-        if (model->wallet().getDefaultAddressType() == OutputType::BECH32) {
-            ui->useBech32->setCheckState(Qt::Checked);
-        } else {
-            ui->useBech32->setCheckState(Qt::Unchecked);
+        // Populate address type dropdown and select default
+        ui->addressType->insertItem((int)OutputType::LEGACY, "Base58 (Legacy)");
+        ui->addressType->setItemData((int)OutputType::LEGACY, "Not recommended due to higher fees and less protection against typos.", Qt::ToolTipRole);
```

Comment:
It becomes a problem when the user sets `-addresstype=legacy`, because then nothing would be selected by default. Previously the checkbox would toggle between legacy and bech32 in that case.

Reply:
When unchecked, `-addresstype=legacy` would map to `p2sh-witness` in the GUI, not `legacy`. So I think we should preserve that, no?

Reply:
I think Bitcoin Core users are already above average Bitcoin users and mostly know what they are doing, don't think we should limit people from creating legacy P2PKH addresses, if they want to.

Reply:
No one has been using this for the past 4 years, and I am not aware of anyone complaining about the lack of it? So it seems odd to add an option that has no use case.

Reply:
I have used this by using RPC. Would have used in GUI if it would have been available.

Reply:
Correction, the checkbox is not disabled on master when using  `-addresstype=legacy`. The original PR description was wrong (or the behavior was changed later).

On master you can test that, when launched with  `-addresstype=legacy`, it would produce a P2PKH address, not wrapped segwit. This happened in the code below:

<img width="540" alt="Schermafbeelding 2021-12-10 om 19 45 45" src="https://user-images.githubusercontent.com/10217/145576223-4cee1254-ceb6-4a38-a5a5-1a5fd9112c9e.png">